### ISSUE TITLE
RedirectRoute not working on Android

### ISSUE THREAD

**Issue by Aqua-Ye** [Milad-Akarie/auto_route_library#1687]
I am using auto route 7.8.0.

I have routes defined in `app_router.dart` like this:
```
        RedirectRoute(
          path: '/upcoming',
          redirectTo: '/shows/upcoming',
        ),
        CustomRoute(
          path: '/',
          page: MainRoute.page,
          guards: [AuthGuard()],
          children: [
            CustomRoute(
              path: 'shows',
              page: SeriesPageRoute.page,
              children: [
                CustomRoute(
                  path: 'watchlist',
                  page: SeriesWatchlistRoute.page,
                ),
                CustomRoute(
                  path: 'upcoming',
                  page: SeriesUpcomingRoute.page,
                ),
              ],
            ),
            ...other tabs...
```

On Android, when I open the deep link `/upcoming`, the shows>upcoming tab opens fine, but if I try to use `/shows/upcoming` directly, it doesn't open the tab at all.
On iOS both links work fine...

Am I doing something wrong?

**Comment by Aqua-Ye** [comment]
in `route_information_parser.dart`, `final uri = Uri.parse(routeInformation.location ?? '');` is losing the first component of the route i'm trying to reach. I believe this is a bug.

For example if I try to navigate to `/shows/upcoming` by calling `appRouter.navigateNamed`, `deeplinkBuilder` callback only provides `/upcoming`.

**Comment by Aqua-Ye** [comment]
OK so in fact there was a breaking change in flutter 3.13.0 - see https://docs.flutter.dev/release/breaking-changes/route-information-uri

**Comment by Milad-Akarie** [comment]
@Aqua-Ye just pushed a new release with refactored code that uses RouteInformation.uri instead of .location, can you check if you still have a problem?

**Comment by Aqua-Ye** [comment]
@Milad-Akarie Thanks. In fact the problem was in my code. Because I enabled `flutter_deeplinking_enabled` in `AndroidManifest.xml`, the `deeplinkBuilder` was getting called in addition to my `uriLinkStream` handler - and that was causing conflicts I guess. 

With your latest change, deeplinkBuilder is now called with the full deep link, but the navigation is broken. I have created a sample app to show the broken behaviour. See https://github.com/Aqua-Ye/auto_route_redirect.

Steps to reproduce:
- navigate to Profile tab
- try to open deep link: `adb shell 'am start -W -a android.intent.action.VIEW -c android.intent.category.BROWSABLE -d "myapp://shows/upcoming"'`
=> it should navigate to the Shows>Upcoming screen

**Comment by Aqua-Ye** [comment]
@Milad-Akarie In fact my original issue was because I enabled `flutter_deeplinking_enabled` in `AndroidManifest.xml`. So it was going through `deeplinkBuilder` and my custom `uriLinkStream` handler.

However about your new release, now the full deep link is passed to the `deeplinkBuilder`, but the navigation is broken.
Check out my sample project https://github.com/Aqua-Ye/auto_route_redirect.

Steps to reproduce:
- on Android
- open Profile tab
- open deep link: `adb shell 'am start -W -a android.intent.action.VIEW -c android.intent.category.BROWSABLE -d "myapp://shows/upcoming"'`
=> the Shows>Upcoming tab should open

**Comment by Milad-Akarie** [comment]
@Aqua-Ye when your link "myapp://shows/upcoming"  is parsed by `Uri.parse` "shows"  is considered a host, not a path segment so if you print Uri.parse('myapp://shows/upcoming').path the output is '/upcoming';
in such cases auto-route needs to check if the URI has a valid host "domain" otherwise treat the host as a part of the path. 
this fix is included in v7.8.3 kindly, confirm if it works for you then close the issue

**Comment by Aqua-Ye** [comment]
> @Aqua-Ye when your link "myapp://shows/upcoming" is parsed by `Uri.parse` "shows" is considered a host, not a path segment so if you print Uri.parse('myapp://shows/upcoming').path the output is '/upcoming'; in such cases auto-route needs to check if the URI has a valid host "domain" otherwise treat the host as a part of the path. this fix is included in v7.8.3 kindly, confirm if it works for you then close the issue

Looks good!
